src/app/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,171 @@
+"use client";
+import * as React from "react";
+import {
+  AppBar,
+  Box,
+  CssBaseline,
+  Divider,
+  Drawer,
+  IconButton,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+import { HiOutlineLogout } from "react-icons/hi";
+import logo from "@/Assets/cha-translate-2-svgrepo-com.svg";
+import Image from "next/image";
+import { AiOutlineMenu } from "react-icons/ai";
+import styles from "./style.module.scss";
+import DialogNewProject from "../dialog NewProject/DialogNewProject";
+import Listsprojects from "../list Projects/Listsprojects";
+import Cookies from "js-cookie";
+import { useRouter } from "next/navigation";
+import { useQuery } from "react-query";
+import { getAllProjects } from "@/app/service/ProjectServices";
+
+const drawerWidth = 240;
+export default function ResponsiveDrawer() {
+  const router = useRouter();
+  const [mobileOpen, setMobileOpen] = React.useState(false);
+  const [isClosing, setIsClosing] = React.useState(false);
+
+  const handleDrawerClose = () => {
+    setIsClosing(true);
+    setMobileOpen(false);
+  };
+
+  const handleDrawerTransitionEnd = () => {
+    setIsClosing(false);
+  };
+
+  const handleDrawerToggle = () => {
+    if (!isClosing) {
+      setMobileOpen(!mobileOpen);
+    }
+  };
+
+  const handleLogout = () => {
+    Cookies.remove("authToken");
+    router.push("/sign-in");
+  };
+
+  const {
+    data: Allprojects,
+    isLoading,
+    error,
+  } = useQuery("Allprojects", getAllProjects);
+  // if (isLoading)
+  //   return (
+  //     <Typography variant="h2">
+  //       Loading projects <AiOutlineLoading3Quarters />{" "}
+  //     </Typography>
+  //   );
+  // if (error)
+  //   return (
+  //     <Typography variant="h2">
+  //       Error loading projects <BiError color="yellow" />
+  //     </Typography>
+  //   );
+
+  const drawerContent = (
+    <Box className={styles.SideBar}>
+      <Box className={styles.LogoContainer}>
+        <Image src={logo} width={40} alt="logo" />
+        <Typography variant="h6">Translator</Typography>
+      </Box>
+      <Divider />
+
+      <DialogNewProject />
+
+      <Divider />
+
+      <Listsprojects AllprojectsProps={Allprojects} />
```

Comment:
you only use allprojects here , so why you do that just take the query and put it into list projects , why you make it a Prop??
note : `Alllprojects` wrong should be `allProjects`